### ISSUE TITLE
Test the model from the Kinetics pre-trained resnet50/101 ckpt: got 0 accuracy on kinetics validation set

### ISSUE THREAD

**Issue by shuchenweng** [kenshohara/3D-ResNets-PyTorch#102]
With pytorch0.3.0 and python3, 
I am trying the validation.py with provided pre-trained resnet50/101 ckpts, to see its performance on kinetics validation set, and I got the zero accuracy, here is the details w.r.t. resnet101 (same things happens to resnet50)

python3 main.py --root_path /data/scweng/kinetics/data --video_path kinetics_videos/jpg --annotation_path kinetics.json \
> --result_path results --pretrain_path resnet-101-kinetics.pth --dataset kinetics --model resnet \
> --model_depth 101 --n_classes 400 --batch_size 128 --n_threads 4 --checkpoint 5
loading pretrained model /data/scweng/kinetics/data/resnet-101-kinetics.pth
dataset loading [0/19906]
dataset loading [1000/19906]
*******
dataset loading [18000/19906]
dataset loading [19000/19906]
run
validation at epoch 1
Epoch: [1][1/468]	Time 42.671 (42.671)	Data 5.162 (5.162)	Loss 6.0356 (6.0356)	Acc 0.000 (0.000)
Epoch: [1][2/468]	Time 0.306 (21.489)	Data 0.000 (2.581)	Loss 6.0250 (6.0303)	Acc 0.000 (0.000)
Epoch: [1][3/468]	Time 0.304 (14.427)	Data 0.000 (1.721)	Loss 6.0094 (6.0233)	Acc 0.000 (0.000)
Epoch: [1][4/468]	Time 0.303 (10.896)	Data 0.000 (1.291)	Loss 6.0002 (6.0176)	Acc 0.000 (0.000)
Epoch: [1][5/468]	Time 0.304 (8.778)	Data 0.000 (1.033)	Loss 6.0275 (6.0195)	Acc 0.000 (0.000)
Epoch: [1][6/468]	Time 0.304 (7.366)	Data 0.000 (0.861)	Loss 6.0734 (6.0285)	Acc 0.016 (0.003)
Epoch: [1][7/468]	Time 0.303 (6.357)	Data 0.000 (0.738)	Loss 6.0240 (6.0279)	Acc 0.000 (0.002)
Epoch: [1][8/468]	Time 0.376 (5.609)	Data 0.000 (0.646)	Loss 5.9835 (6.0223)	Acc 0.000 (0.002)
Epoch: [1][9/468]	Time 0.302 (5.019)	Data 0.000 (0.574)	Loss 6.1109 (6.0322)	Acc 0.000 (0.002)
Epoch: [1][10/468]	Time 2.656 (4.783)	Data 2.235 (0.740)	Loss 6.0769 (6.0366)	Acc 0.000 (0.002)

I have double checked my kinetics data and just used the project and pretrained ckpts from the github, Could you help me about this?

**Comment by shuchenweng** [comment]
it turns out we use the finetune mode for evaluation, which means that only pretrained convs weights are  loaded, and the fully-connected layers are random initialized, hence the 0.0 accuracy on validation. After fixing this problem, we obtained similar results as reported.

**Comment by LiJinyao** [comment]
> it turns out we use the finetune mode for evaluation, which means that only pretrained convs weights are loaded, and the fully-connected layers are random initialized, hence the 0.0 accuracy on validation. After fixing this problem, we obtained similar results as reported.

How do you fix this ?

**Comment by forwchen** [comment]
To use the pretrained model directly for testing, you need to comment out the code that changes the models' fc layer for finetuning, namely these two if-else blocks:
https://github.com/kenshohara/3D-ResNets-PyTorch/blob/d9847ea56b0d4bc7601a229099deb2b483539514/model.py#L175

https://github.com/kenshohara/3D-ResNets-PyTorch/blob/d9847ea56b0d4bc7601a229099deb2b483539514/model.py#L194

**Comment by HaiyiMei** [comment]
@forwchen @shuchenweng  Hey guys, I followed the step to comment out two if-else blocks. But I got a result that is much lower than reported. I got 50% in resnet-101, 48% in resnet-50.

I'm running the code in pytorch 1.0.0 / CUDA 9, maybe it's the cause? But my GPU is RTX 2070, which doesn't support the lower vision than CUDA 9. Do you guys ever test it in pytorch 1.x?
